operations/observability/Observability.md

```diff
@@ -0,0 +1,318 @@
+
+- Multi-interval continuous health checks.
+  This defines and implements regular scheduled health checks to occur both during and after the installation and upgrade.
+  We can select the appropriate time interval in the Grafana dashboard to check the results for past six hours, a day, and a week.
+
+- Automate the Observability framework , installation, and configuration of the components involved in the Install and Upgrade Observability Framework in a CSM environment.
+
+- The automatic generation of configurable Grafana dashboards that provide key insights and Key Performance Indicators (KPIs).
+  These dashboards show the frequency of errors across the complex systems, include panels to visualize the outliers, and provide a way to identify the trends in the complex system across different dimensions.
+
+The following IUF topics are discussed in the following subsections.
+
+- [Automation of Observability Framework](#automation-of-observability-framework)
+  - [Features](#observability-framework-features)
+  - [Observability workflow](#observability-workflow)
+  - [`systemd` services](#systemd-services)
+- [IUF timing dashboard](#iuf-timing-dashboard)
+  - [Features](#timing-dashboard-features)
+  - [Prometheus metrics using Argo workflow](#prometheus-metrics-using-argo-workflow)
+  - [Timing dashboard](#timing-dashboard)
+- [Goss tests for PIT and NCN](#goss-tests-for-pit-and-ncn)
+  - [Overview](#goss-test-overview)
+  - [Workflow](#workflow)
+  - [Log file format](#log-file-format)
+  - [Grok-exporter deployment, service, and service-monitor](#grok-exporter-deployment-service-and-service-monitor)
+  - [Configuration file for the grok-exporter](#configuration-file-for-the-grok-exporter)
+  - [Prometheus metrics and Grafana dashboard](#prometheus-metrics-and-grafana-dashboard)
+- [Error dashboard](#error-dashboard)
+  - [Features](#error-dashboards-features)
+  - [Error dashboards](#error-dashboards)
+
+## Automation of Observability Framework
+
+Grok-exporter, Prometheus, and Grafana are instantiated automatically on the PIT node.
+
+### Observability framework features
+
+- Three different services for Prometheus, grok-exporter, and Grafana will be automatically running as a part of PIT node installation.
+- Prometheus, grok-exporter, and Grafana containers are deployed with services initially on the PIT node for monitoring before other products are installed.
+- Grok-exporter parses the unstructured data from the log files and creates Prometheus metrics.
+- Prometheus is used as a time-series database for capturing the metrics.
+- All the Grafana dashboards automatically load up.
+
+### Observability workflow
+
+![Observability framework workflow](../../img/operations/AutomationFrameworkWorkflow.png "Observability framework workflow")
+
+### `systemd` services
+
+Running status of the PIT observability services: grok-exporter, Prometheus, and Grafana.
```

Comment:
```suggestion
If problems are suspected with the Observability Framework during a CSM install, an administrator may that the
grok-exporter, Prometheus, and Grafana services are running properly on the PIT node.
```

Reply:
This change request is not correct. The sentence is partial.